File: ecommerce-service/src/main/java/com/ceygreen/ecommerce/entity/Product.java
package com.ceygreen.ecommerce.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "farmer_id", nullable = false, columnDefinition = "uuid")
    private UUID farmerId;

    @Column(name = "crop_name", nullable = false)
    private String cropName;

    @Column(nullable = false)
    private int quantity;

    @Column(name = "unit_price", nullable = false, precision = 12, scale = 2)
    private BigDecimal unitPrice;

    @Column(name = "harvest_date", nullable = false)
    private LocalDate harvestDate;

    @Column(nullable = false)
    private String location;

    @Column(columnDefinition = "text")
    private String description;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt = Instant.now();

    @Column(nullable = false)
    private boolean active = true;

    @Version
    private long version;

    public Product() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public UUID getFarmerId() { return farmerId; }
    public void setFarmerId(UUID farmerId) { this.farmerId = farmerId; }
    public String getCropName() { return cropName; }
    public void setCropName(String cropName) { this.cropName = cropName; }
    public int getQuantity() { return quantity; }
    public void setQuantity(int quantity) { this.quantity = quantity; }
    public BigDecimal getUnitPrice() { return unitPrice; }
    public void setUnitPrice(BigDecimal unitPrice) { this.unitPrice = unitPrice; }
    public LocalDate getHarvestDate() { return harvestDate; }
    public void setHarvestDate(LocalDate harvestDate) { this.harvestDate = harvestDate; }
    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }
    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }
    public boolean isActive() { return active; }
    public void setActive(boolean active) { this.active = active; }
    public long getVersion() { return version; }
    public void setVersion(long version) { this.version = version; }
}

File: frontend/src/types/product.ts
export type ProductListingStatus = 'ACTIVE' | 'INACTIVE' | 'OUT_OF_STOCK';

export interface Product {
  id: number;
  farmerId: string;
  cropName: string;
  quantity: number;
  unitPrice: number;
  harvestDate: string;
  location: string;
  description?: string | null;
  imageUrl?: string | null;
  createdAt?: string;
  active: boolean;
  status: ProductListingStatus;
}

export interface ProductCreateRequest {
  cropName: string;
  quantity: number;
  unitPrice: number;
  harvestDate: string;
  location: string;
  description?: string;
  imageUrl?: string;
}

export interface ProductUpdateRequest {
  unitPrice?: number;
  quantity?: number;
  active?: boolean;
  description?: string;
  imageUrl?: string;
  location?: string;
}

export interface ProductListParams {
  q?: string;
  cropName?: string;
  location?: string;
  minPrice?: number;
  maxPrice?: number;
  inStock?: boolean;
  active?: boolean;
  status?: ProductListingStatus;
  sort?: 'price_asc' | 'price_desc' | 'newest';
  page?: number;
  size?: number;
}
